server/util.go

```diff
@@ -371,3 +375,47 @@ func minDuration(a, b time.Duration) time.Duration {
 	}
 	return b
 }
+
+func diffRegionPeersInfo(origin *regionInfo, other *regionInfo) string {
+	var ret []string
+	for _, a := range origin.Peers {
+		both := false
+		for _, b := range other.Peers {
+			if reflect.DeepEqual(a, b) {
+				both = true
+				break
+			}
+		}
+		if !both {
+			ret = append(ret, fmt.Sprintf("Remove peer:{%v}", a))
+		}
+	}
+	for _, b := range other.Peers {
+		both := false
+		for _, a := range origin.Peers {
+			if reflect.DeepEqual(a, b) {
+				both = true
+				break
+			}
+		}
+		if !both {
+			ret = append(ret, fmt.Sprintf("Add peer:{%v}", b))
+		}
+	}
+	return strings.Join(ret, ",")
+}
+
+func diffRegionKeyInfo(origin *regionInfo, other *regionInfo) string {
+	var ret []string
+	if !bytes.Equal(origin.Region.StartKey, other.Region.StartKey) {
+		originKey := &metapb.Region{StartKey: origin.Region.StartKey}
+		otherKey := &metapb.Region{StartKey: other.Region.StartKey}
+		ret = append(ret, fmt.Sprintf("StartKey Changed:%s -> %s", originKey, otherKey))
```

Comment:
it just prints start_key field without `Region`. protoc just expose Marshaler methods for Message. I need to do that.